lib/cli/src/Cardano/CLI.hs

```diff
@@ -501,6 +504,30 @@ decodeKey = \case
       where
         dpErr = "Internal error: Unable to extract bytes from bech32 data part"
 
+        bech32Err  = ("Bech32 error: " <>) . \case
+            StringToDecodeTooLong -> "string is too long"
+            StringToDecodeTooShort -> "string is too short"
+            StringToDecodeHasMixedCase -> "string has mixed case"
+            StringToDecodeMissingSeparatorChar -> "string has no separator char"
+            StringToDecodeContainsInvalidChars [] -> invalidCharsMsg
+            StringToDecodeContainsInvalidChars ixs -> invalidCharsMsg <> ":\n"
+                    <> markCharsRedAtIndices (map unCharPos ixs) (T.unpack t)
+        invalidCharsMsg = "Invalid character(s) in string"
+        unCharPos (CharPosition x) = x
+
+markCharsRedAtIndices :: Integral i => [i] -> String -> String
+markCharsRedAtIndices ixs txt = go 0 ixs txt ++ def
+  where
+    go _c [] [] = mempty
+    go c (i:is) (s:ss)
+        | c == i    = red ++ s:def ++ go (c + 1) is ss
```

Comment:
This will output 2 ansi escape sequences for each invalid character.

Reply:
Yes, but it still works, and multiple invalid characters should rare.